test/integration/users_edit_test.rb

```diff
@@ -0,0 +1,51 @@
+require "test_helper"
+
+class UsersEditTest < ActionDispatch::IntegrationTest
+  include Capybara::DSL
+
+  test "unsuccessful edit" do
+    user = create(:user)
+    get edit_user_path(user)
+    assert_template 'users/edit'
+    patch user_path(user), user: { username: "",
+                                   email: "foo@invalid",
+                                   password: "foo",
+                                   password_confirmation: "bar" }
+    refute flash.empty?
+    assert_template 'users/edit'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.